File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.py
class Solution:
    def buildArray(self, target: List[int], n: int) -> List[str]:
        seen = set(target)
        stack = []
        for i in range(1, n + 1):
            stack.append('Push')
            if not i in seen:
                stack.append('Pop')
            if i == target[-1]:
                break
        return stack

File: 1441-build-an-array-with-stack-operations/1441-build-an-array-with-stack-operations.js
/**
 * @param {number[]} target
 * @param {number} n
 * @return {string[]}
 */
var buildArray = function(target, n) {
  const set = new Set(target);
  const stack = [];
  
  for (let i = 1; i <= n; i++) {
    stack.push('Push');
    if (!set.has(i)) stack.push('Pop');
    if (i == target.at(-1)) break;
  }
  
  return stack;
};